### ISSUE TITLE
Hidden modules shown/loaded differently with/without cache

### ISSUE THREAD

**Issue by zao** [TACC/Lmod#483]
This is Lmod 8.4.11.

Given a module file whose name has a leading period to hide it, I'm observing differences in how `ml spider` displays the module name when invoked with and without `--ignore-cache`.

```
b-an01 [~]$ ml --show_hidden spider GROMACS
<snip>
        GROMACS/5.1.4
        GROMACS/2016.x-drude-20180214-g3f7439a
        GROMACS/2016.2
```
vs.
```
b-an01 [~]$ ml --show_hidden --ignore-cache spider GROMACS
<snip>
        GROMACS/.2016.x-drude-20180214-g3f7439a
        GROMACS/5.1.4
        GROMACS/2016.2
```

Trying to load the module with the cache-sourced name results in an error:
```
b-an01 [~]$ ml GCC/6.4.0-2.28  impi/2017.3.196
b-an01 [~]$ ml GROMACS/2016.x-drude-20180214-g3f7439a
Lmod has detected the following error:  Unable to load module because of error when evaluating modulefile:
     /hpc2n/eb/modules/all/MPI/GCC/6.4.0-2.28/impi/2017.3.196/GROMACS/2016.x-drude-20180214-g3f7439a.lua: Empty or non-existant file
```

Trying to load it with the true hidden name results in the following error:
```
b-an01 [~]$ ml GCC/6.4.0-2.28  impi/2017.3.196
b-an01 [~]$ ml GROMACS/.2016.x-drude-20180214-g3f7439a
Lmod has detected the following error:  The following module(s) are unknown: "GROMACS/.2016.x-drude-20180214-g3f7439a"
<snip>
```
while loading it as `ml --ignore-cache GROMACS/.2016.x-drude-20180214-g3f7439a` succeeds.

**Comment by rtmclay** [comment]
I don't see this in my tests.  I tested both the latest version 8.4.15 and 8.4.11.  Both work correctly.   I suspect the following:

1. Your site is configured for --with-cached_loads=yes (or LMOD_CACHED_LOADS=yes)
2. The cache file got build with GROMACS/2016.x-drude-20180214-g3f7439a
3. Your site moved GROMACS/2016.x-drude-20180214-g3f7439a to GROMACS/.2016.x-drude-20180214-g3f7439a but the cache file was not updated.

You can look your cache file (spiderT.lua) and look for the GROMACS entry.

If I am correct, the fix is to rebuilt the system spider cache file.

**Comment by zao** [comment]
I rebuilt the spider cache before reporting this so it's up to date, but I just found the root cause in our system.

Software (and the system spider cache) are built on a writable filesystem, which is then published onto a read-only filesystem that end-user loads run against.

The module filename differs between the two filesystems where an administrator seems to have unhidden it on the source filesystem but failed to publish the change. Rebuilding the cache thus had no effect as rebuild saw a different filename than in production.

Chalk this one up to overly clever scripts/admins.
